src/RegisterForm.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useContext, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { createUser, getUser } from './api';
+import { FormInput } from './FormInput';
+import { Loader } from './Loader';
+import './styles/LoginForm.scss';
+import { TodosContext } from './TodosProvider';
+import { User } from './types/User';
+
+type Props = {
+  onSetTypeForm: React.Dispatch<React.SetStateAction<boolean>>,
+};
+
+export const RegisterForm: React.FC<Props> = React.memo(({ onSetTypeForm }) => {
+  const [errorUsername, setErrorUsername] = useState(false);
+  const [{
+    username, name, email, phone,
+  }, setUser] = useState<User>({
+    username: '',
+    name: '',
+    email: '',
+    phone: '',
+  });
+
+  const [isLoad, setIsLoad] = useState(false);
+
+  const [isError, setIsError] = useState(false);
+
+  const { setTodos, setUserId } = useContext(TodosContext);
+
+  const navigate = useNavigate();
+
+  const handlerRegister = () => {
+    setIsLoad(true);
+    getUser(username).then(users => {
+      if (users.length < 1) {
+        throw new Error('No user');
+      }
+
+      setErrorUsername(true);
+      setIsLoad(false);
+    })
+      .catch(() => {
+        if (name && phone && email) {
+          setIsError(false);
+          createUser(name, username, email, phone).then(createdUser => {
+            setUserId(createdUser.id);
+            navigate(`/${createdUser.username}/`);
+            setIsLoad(true);
+            setTodos([]);
+          });
+        } else {
+          throw new Error('Failed to create user');
+        }
+      })
+      .catch(() => {
+        setErrorUsername(true);
+        setIsLoad(false);
+      });
+  };
+
+  return (
+    <>
+      {!isLoad ? (
+        <div className="form">
+          <h2 className="form__title">
+            Account login
```

Comment:
```suggestion
        }
      })
      .catch(() => {
        setErrorUsername(true);
        setIsLoad(false);
      });
  };
  
  if (isLoad) {
    return <Loader />
  }

  return (
    <>
      {!isLoad ? (
        <div className="form">
          <h2 className="form__title">
            Account login
```
better to manage this in such way
instead of such big ternary operator